components/dialog/test/dialog-confirm.vdiff.js

```diff
@@ -0,0 +1,75 @@
+import '../../button/button.js';
+import '../dialog-confirm.js';
+import { expect, fixture, html } from '@brightspace-ui/testing';
+
+const buttons = html`
+	<d2l-button slot="footer" primary>Yes</d2l-button>
+	<d2l-button slot="footer" id="cancel">No</d2l-button>
+`;
+
+const confirmDialog = html`
+	<d2l-dialog-confirm id="confirm" title-text="Title" text="Are you sure?" opened>
```

Comment:
Setting up by using `opened` attribute is definitely much cleaner, which is great. I think the helper was implicitly also testing the `d2l-dialog-open` & `d2l-dialog-close` events. I think the open event may still be covered by some unit tests (also implicitly), but I'm not sure the close event is covered. I'm wondering if it would be a good idea for us to add an explicit unit test for those events, since consumers do rely on them. Or maybe you already checked and found they are covered?

Reply:
So I'd say that's _kinda_ covered by the [dialog-mixin.vdiff.js](https://github.com/BrightspaceUI/core/blob/main/components/dialog/test/dialog-mixin.vdiff.js) tests.  But I like the idea of adding non-vdiff, dedicated tests - I'll add some to this PR

Reply:
If we do, I'm thinking those unit tests should be for the general mixin case.

Reply:
Ok, just pushed a new file - it'll fail until we expose `oneDefaultPreventedEvent` in `@brightspace-ui/testing`, but is that what you had in mind?

Reply:
Ok, merged the change in `@brightspace-ui/testing` and now CI here is passing.

Reply:
Yep, exactly. Looks great!